autoload/crystalline.vim

```diff
@@ -382,7 +440,13 @@ function! crystalline#sep(group_a, group_b, ch, left) abort
   if a:left == 0 && a:group_a ==# 'TabType' && index(get(g:, 'crystalline_tab_type_fake_separators', []), a:group_b) >= 0
     let l:sep_item = g:crystalline_tab_separator
   else
-    let l:sep_item = '%#Crystalline' . crystalline#get_sep_group(a:group_a, a:group_b) . '#' . a:ch
+    let l:sep_group = crystalline#get_sep_group(a:group_a, a:group_b)
+    " Create if it doesn't exist
+    if !hlexists(l:sep_group)
```

Comment:
Alright, so I figured it out. The current implementation is only working because `hlexists` always returns true here. You're checking for ex. `TabTypeToInsertMode`, not `CrystallineTabTypeToInsertMode`, so the highlight group always gets re-generated. Fixing this seems to break the whole implementation, not just changing the theme.

Reply:
Ugh, dammit. Alright, let's see what we can salvage.